packages/nimble-components/src/table/components/row/index.ts

```diff
@@ -80,31 +100,7 @@ export class TableRow<
     // https://github.com/microsoft/fast/issues/5750
     private ignoreSelectionChangeEvents = false;
 
-    @volatile
-    public get columnStates(): ColumnState[] {
-        return this.columns.map((column, i) => {
-            const fieldNames = column.columnInternals.dataRecordFieldNames;
-            let cellState: TableCellState | undefined;
-            if (this.hasValidFieldNames(fieldNames) && this.dataRecord) {
-                const cellDataValues = fieldNames.map(
-                    field => this.dataRecord![field]
-                );
-                const cellRecord = Object.fromEntries(
-                    column.columnInternals.cellRecordFieldNames.map((k, j) => [
-                        k,
-                        cellDataValues[j]
-                    ])
-                );
-                const columnConfig = column.columnInternals.columnConfig;
-                cellState = {
-                    cellRecord,
-                    columnConfig
-                };
-            }
-            const cellIndentLevel = i === 0 && this.nestingLevel > 0 ? this.nestingLevel - 1 : 0;
-            return { column, cellState, cellIndentLevel };
-        });
-    }
+    private columnNotifiers?: Notifier[] = [];
```

Comment:
Does `columnNotifiers` need to be nullable? It looks like it always gets assigned a (possibly empty) array, not null or undefined.

Reply:
This is an interesting thing we've run into before -- the initial value of `[]` gets set in the constructor, along with all the other initial values. After each class member is initialized, its change handler is run before initializing the next class member & running its change handler. As a result, the `columns` change handler is called before `columnNotifiers` is initialized, which means that `removeColumnObservers` is called while `columnNotifiers` is still `undefined`. Therefore, I marked `columnNotifiers` with `?` since there are code paths that occur with `undefined` as its value.

One approach we've used in some places to avoid this problem is to look at the value of `this.$fastController.isConnected` in the change handler to not run the bulk of its implementation before the element is connected and then make sure everything is in the correct state in the `connectedCallback` since we hadn't been handling changes before then. Maybe that would be a better approach here as well. Any thoughts?

Reply:
Ah, great explanation. It is definitely a mindset change to realize that something that appears to be initialized when it is defined is actually initialized after some of our code may have already run.

I'm not feeling a strong opinion either way. I don't think the broader type here is enough of a problem to justify much refactoring. But maybe if this keeps coming up we should form a policy around it and try to be consistent. For now I'm personally ok to leave it as-is.

Reply:
The other approach I've been pushing is to make it so the property initialization happens in the correct order to prevent change handler issues (and that may be a good idea here too).

That would mean making it public internal (to appease the lint rules) and moving `columnNotifiers` property initialization before `columns` property initialization. That way on columns property initialization and change handler execution `columnNotifiers` should be ready to go.

I kinda want to move away from the usage of isConnected except only for the case of avoiding unnecessary view rendering work until it is needed. Anything that can effect the state of the control should run even if not connected to the DOM. i.e. document.createElement and other imperative apis should behave well and properties on the element should report accurate info even if not attached to the DOM (same as native elements).